Content.Shared/_EinsteinEngines/Revolutionary/RevolutionaryConverterSystem.cs

```diff
@@ -0,0 +1,115 @@
+using Content.Shared._EinsteinEngines.Revolutionary.Components;
+using Content.Shared.Chat;
+using Content.Shared.Dataset;
+using Content.Shared.DoAfter;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Random.Helpers;
+using Content.Shared.Revolutionary.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Shared._EinsteinEngines.Revolutionary;
+
+public sealed class RevolutionaryConverterSystem : EntitySystem
+{
+    private const string RevConvertSpeechBaseKey = "revolutionary-converter-speech-";
+
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] private readonly SharedChatSystem _chat = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    private LocalizedDatasetPrototype? _speechLocalization;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RevolutionaryConverterComponent, RevolutionaryConverterDoAfterEvent>(OnConvertDoAfter);
+        SubscribeLocalEvent<RevolutionaryConverterComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<RevolutionaryConverterComponent, AfterInteractEvent>(OnConverterAfterInteract);
+
+        _speechLocalization = _prototypeManager.Index<LocalizedDatasetPrototype>("revolutionary_converter_speech");
```

Comment:
And use that localized dataset ID here
```suggestion
        _speechLocalization = _prototypeManager.Index<LocalizedDatasetPrototype>(RevConvertSpeechProto);
```